econml/solutions/causal_analysis/_causal_analysis.py

```diff
@@ -1136,10 +1274,120 @@ def _heterogeneity_tree_string(self, Xtest, feature_index, *,
+            values and the default value (i.e., if non-scalar the array should have shape (n,d_t-1))
+        alpha: float in [0, 1], default 0.1
+            Confidence level of the confidence intervals
+            A (1-alpha)*100% confidence interval is returned
+
+        Returns
+        -------
+        output: DataFrame
+            Dataframe containing recommended treatment, effect, confidence interval, sorted by effect
+        """
+        result = self._safe_result_index(Xtest, feature_index)
+
+        # get dataframe with all but selected column
+        orig_df = pd.DataFrame(Xtest, columns=self.feature_names_).drop(self.feature_names_[result.feature_index],
+                                                                        axis=1)
+
+        Xtest = result.X_transformer.transform(Xtest)
+        if Xtest.shape[1] == 0:
+            Xtest = None
+
+        if result.feature_baseline is None:
+            # apply 10% of a typical treatment for this feature
+            effect = result.estimator.effect_inference(Xtest, T1=result.treatment_value * 0.1)
+        else:
+            effect = result.estimator.const_marginal_effect_inference(Xtest)
+
+        effect.translate(-treatment_costs)
+
+        est = effect.point_estimate
+        est_lb = effect.conf_int(alpha)[0]
+        est_ub = effect.conf_int(alpha)[1]
+
+        if result.feature_baseline is None:
+            rec = np.empty(est.shape[0], dtype=object)
+            rec[est > 0] = "increase"
+            rec[est <= 0] = "decrease"
+            # set the effect bounds; for positive treatments these agree with
+            # the estimates; for negative treatments, we need to invert the interval
+            eff_lb, eff_ub = est_lb, est_ub
+            eff_lb[est <= 0], eff_ub[est <= 0] = -eff_ub[est <= 0], -eff_lb[est <= 0]
+            # the effect is now always positive since we decrease treatment when negative
+            eff = np.abs(est)
+        else:
+            # for discrete treatment, stack a zero result in front for control
+            zeros = np.zeros((est.shape[0], 1))
+            all_effs = np.hstack([zeros, est])
+            eff_ind = np.argmax(all_effs, axis=1)
+            all_eff_lbs = np.hstack([zeros, est_lb])
+            all_eff_ubs = np.hstack([zeros, est_ub])
+            treatment_arr = np.array([result.feature_baseline] + [lvl for lvl in result.feature_levels], dtype=object)
+            rec = treatment_arr[eff_ind]
+            eff_ind = eff_ind.reshape(-1, 1)
+            eff = np.take_along_axis(all_effs, eff_ind, 1).reshape(-1)
+            eff_lb = np.take_along_axis(all_eff_lbs, eff_ind, 1).reshape(-1)
+            eff_ub = np.take_along_axis(all_eff_ubs, eff_ind, 1).reshape(-1)
+
+        df = pd.DataFrame({'Treatment': rec,
+                           'Effect of treatment': eff,
+                           'Effect of treatment lower bound': eff_lb,
```

Comment:
Out of scope of PR, but you may want to make these constants publicly accessible. Maybe just `CausalAnalysis.TREATMENT_EFFECT_LOWER`? Then EconML users don't have to redefine their own constants.